pkg/apiserver/event/event_test.go

```diff
@@ -0,0 +1,349 @@
+					Message:      "test",
+					StartTime:    nil,
+					FinishTime:   nil,
+					Duration:     "1m",
+					Pods:         nil,
+					ExperimentID: "testUID",
+				},
+			}
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/dry?kind=testKind", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Code).Should(Equal(http.StatusOK))
+			responseBody, err := json.Marshal(response)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Body.Bytes()).Should(Equal(responseBody))
+		})
+
+		It("test err", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/dry?kind=err", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Code).Should(Equal(http.StatusInternalServerError))
+		})
+	})
+
+	Context("GetEvent", func() {
+		It("success", func() {
+			response := &core.Event{
+				ID:           0,
+				CreatedAt:    time.Time{},
+				UpdatedAt:    time.Time{},
+				DeletedAt:    nil,
+				Experiment:   "testExperiment",
+				Namespace:    "testNamespace",
+				Kind:         "testKind",
+				Message:      "test",
+				StartTime:    nil,
+				FinishTime:   nil,
+				Duration:     "1m",
+				Pods:         nil,
+				ExperimentID: "testUID",
+			}
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/get?id=0", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Code).Should(Equal(http.StatusOK))
+			responseBody, err := json.Marshal(response)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Body.Bytes()).Should(Equal(responseBody))
+		})
+
+		It("empty id", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/get", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Code).Should(Equal(http.StatusBadRequest))
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("bad id", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/get?id=badID", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Code).Should(Equal(http.StatusBadRequest))
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("not found", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/get?id=1", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Code).Should(Equal(http.StatusInternalServerError))
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("other err", func() {
```

Comment:
what is the difference between "other err" and the "not found" above?

Reply:
In API processing, if the error is "Gorm recordnotfounderror", the returned results are different, so there are two cases.